app/authorization.go

```diff
@@ -147,8 +147,10 @@ func (a *App) HasPermissionToTeam(askingUserId string, teamId string, permission
 		return false
 	}
 	teamMember, _ := a.GetTeamMember(teamId, askingUserId)
-	if teamMember != nil && teamMember.DeleteAt == 0 {
-		return a.RolesGrantPermission(teamMember.GetRoles(), permission.Id)
+	if teamMember != nil {
```

Comment:
Doesn't this allows a user to list the channels of team that she was formerly a member of (i.e. after being removed or leaving a team)?

So a user still has permissions on a team after they leave it.

Reply:
Ah. I read how the test was working wrong. Then I removed that check because I thought the test passed without it (thinking that it wouldn't retrieve a deleted team member.)
Fixed it and the test. Good catch!